identity/signer.go

```diff
@@ -0,0 +1,44 @@
+package identity
+
+import (
+	"github.com/ethereum/go-ethereum/crypto"
+	"fmt"
+)
+
+type Signer interface {
+	Sign(message []byte) ([]byte, error)
+}
+
+type keystoreSigner struct {
+	keystoreManager keystoreInterface
+	identity        Identity
+}
+
+func NewSigner(keystore keystoreInterface, identity Identity) Signer {
+	return &keystoreSigner{
+		keystoreManager: keystore,
+		identity:        identity,
+	}
+}
+
+func (ksSigner *keystoreSigner) Sign(message []byte) ([]byte, error) {
+	signature, err := ksSigner.keystoreManager.SignHash(identityToAccount(ksSigner.identity), messageHash(message))
```

Comment:
If I am signing 100 messages, same `identityToAccount` is called several times for variable `self.identity` is is always same